### PR TITLE
Fix default datetime values for test result upload - covid pipeline

### LINKED ISSUE #7087: Investigate collection date time is less than result date time for CO
Previous effort:
https://github.com/CDCgov/prime-simplereport/pull/6961

### PR DESCRIPTION
**Pull request by emyl3**
# BACKEND PULL REQUEST

## Related Issue

Resolves #7087

## Changes Proposed
- Updates the default timezone for the following dates `testResultDate`, `orderTestDate`, `dateResultReleased` to use the ordering provider address if none is provided for the test result bulk upload - covid pipeline.

## Additional Information
- This is to fix an issue that CO was experiencing:

> The issue happens because the new system does a validation check for incoming records. It requires that the collection date and time (OBR7) is strictly less than the result date and time (OBX19) which also must be strictly less than the reporting date and time (MSH7).

A previous effort (#6961) only addressed the FHIR pipeline. This work addresses the covid pipeline.

## Testing
Deployed on dev2
A [branch with a log statement](https://github.com/CDCgov/prime-simplereport/compare/main...elisa/add-log-test-timestamp) has been added if you want to see what the transformed row will look like.
Add a unique comment to the "Test comment" row so you can easy search the logs for this 😄

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/service/AddressValidationService.java [hunk 1]
```diff
@@ -138,7 +138,7 @@ public ZoneId getZoneIdByLookup(Lookup lookup) {
     try {
       timezoneInfo = getTimezoneInfoByLookup(lookup);
     } catch (InvalidAddressException | IllegalGraphqlArgumentException exception) {
-      log.error("Unable to find timezone by testing lab address", exception);
+      log.error("Unable to find timezone with provided address", exception);
```
Comment (emyl3) [inline backend/src/main/java/gov/cdc/usds/simplereport/service/AddressValidationService.java:141]: ⚠️ Updated to make it more generic

File: backend/src/test/java/gov/cdc/usds/simplereport/service/TestResultUploadServiceTest.java [hunk 2]
```diff
@@ -708,6 +710,63 @@ void uploadService_processCsv_handlesEscapedCommas_inTestingLabValues() {
...
+    sut.processResultCSV(input);
+
+    // THEN
+    TestResultRow row = getRowFromUpload(dataHubMock);
+    assertThat(row.getTestResultDate().getValue()).isEqualTo("2021-12-23T12:00-08:00");
+    assertThat(row.getOrderTestDate().getValue()).isEqualTo("2021-12-20T12:00-08:00");
+    assertThat(row.getSpecimenCollectionDate().getValue()).isEqualTo("2021-12-20T12:00-08:00");
+    assertThat(row.getTestingLabSpecimenReceivedDate().getValue())
+        .isEqualTo("2021-12-20T12:00-08:00");
+    assertThat(row.getDateResultReleased().getValue()).isEqualTo("2021-12-23T12:00-08:00");
```
Comment (mehansen) [inline backend/src/test/java/gov/cdc/usds/simplereport/service/TestResultUploadServiceTest.java:739]: would these fail when we enter DST?

Comment (emyl3) [reply backend/src/test/java/gov/cdc/usds/simplereport/service/TestResultUploadServiceTest.java:739]: ooo that's a good question -- let me investigate 👀

File: backend/src/test/java/gov/cdc/usds/simplereport/utils/BulkUploadResultsToFhirTest.java [hunk 3]
```diff
@@ -538,13 +539,14 @@ void convertExistingCsv_populatesBlankFields() {
 
   @Test
   void convertExistingCsv_setsOrderingProviderTimezone_forNoTimezoneInDates() {
-    when(resultsUploaderCachingService.getZoneIdByAddress(any()))
-        .thenReturn(ZoneId.of("US/Mountain"));
+    ZoneId zoneId = ZoneId.of("US/Mountain");
+    when(resultsUploaderCachingService.getZoneIdByAddress(any())).thenReturn(zoneId);
     InputStream input = loadCsv("testResultUpload/test-results-upload-valid-dates-no-timezone.csv");
 
-    Instant orderTestDate = Instant.parse("2021-12-19T12:00:00-07:00");
```
Comment (emyl3) [inline backend/src/test/java/gov/cdc/usds/simplereport/utils/BulkUploadResultsToFhirTest.java:545]: ⚠️ Went back and changed this in this file, @mehansen, based on your previous comment about these tests failing when it is DST 😅 Thank you for catching that!

File: backend/src/test/java/gov/cdc/usds/simplereport/service/TestResultUploadServiceTest.java (hunk 2 above)
Comment (emyl3) [reply backend/src/test/java/gov/cdc/usds/simplereport/service/TestResultUploadServiceTest.java]: Changed the tests -- let me know what you think!! Thank you for catching that :)

**Comment by emyl3** [comment]
@mehansen ready for re-review!

**Review by mehansen — APPROVED** [review]
so I was playing around with dates that are during DST and the time zone converter applies the correct UTC offset according to whether that date is in DST or not (so not impacted by the date that we're running the test). which is the way it should be and makes sense - sorry for misleading you! 🙈 I think the tests would've been fine before but still looks good now! thanks for finding the cause of CO's bug

**Comment by emyl3** [comment]
> so I was playing around with dates that are during DST and the time zone converter applies the correct UTC offset according to whether that date is in DST or not (so not impacted by the date that we're running the test). which is the way it should be and makes sense - sorry for misleading you! 🙈 I think the tests would've been fine before but still looks good now! thanks for finding the cause of CO's bug

OOO I did not know that hahah -- thank you for testing this out for the future!!! 🙌

**Review by mpbrown — APPROVED** [review]
LGTM! Thanks for working on the time zone issue!

**Pull request merged**